face/capture/src/main/java/com/simprints/face/capture/screens/livefeedback/views/CalculateTargetViewSizeUseCase.kt

```diff
@@ -0,0 +1,50 @@
+package com.simprints.face.capture.screens.livefeedback.views
+
+import android.content.Context
+import com.simprints.core.tools.extentions.dpToPx
+import com.simprints.infra.logging.Simber
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+internal class CalculateTargetViewSizeUseCase @Inject constructor(@ApplicationContext private val context: Context) {
+    companion object{
+        const val BASE_TARGET_SIZE_DP = 240f // Target size in dp
+        const val DEFAULT_MULTIPLIER = 0.5f    // Default multiplier
+    }
+    /**
+     * This function calculates the appropriate width for the target face capture view based on
+     * the device's screen size. The target width is set to half of the smallest dimension
+     * (either the screen width or height), but it will never be smaller than the minimum target
+     * width of 240dp.
+     *
+     * Purpose:
+     * - Ensures that the face capture area adapts to different screen sizes
+     * - Prevents the target view from being too small on large screens by enforcing a minimum
+     *   size, allowing health workers to easily capture a user's face without needing to adjust
+     *   too much based on the device size.
+     */
+
+    operator fun invoke(): Float {
```

Comment:
I had a weird idea for this (not sure if it would be simpler to implement) - would it be possible to enforce the size constraints in XML by using the percentages, ratios and other ConstraintLayout magic to make android do all the calculation and then take the size+position of the view for the crop?

Reply:
@luhmirin-s I attempted to achieve this, but there's no straightforward way to implement everything in XML. I can't create a semi-transparent view that contains a circle with a white stroke, where the circle width equals the screen width/2  or 240dp if the screen width is less than 240dp.

Reply:
@luhmirin-s I restructured the entire handling of this section to simplify it.